proposed/2023/sponsor-link.md

```diff
@@ -0,0 +1,164 @@
+
+![Visual Studio Sponsor Experience](../2023/resources/VisualStudioSponsors.png)
+
+#### dotnet restore
+
+If a developer has the MSBuild property of `<NuGetSponsor>true</NuGetSponsor>` they will see a summary of packages that require funding and an empowering message on how to see the funding needs after a restore.
+
+A simple message will appear at the bottom such as:
+
+```
+X packages are looking for sponsors. Run `dotnet list package --sponsor` for more details.
+```
+
+#### dotnet list package --sponsor
+
+To list all the packages that are looking for sponsors, one can run `dotnet list package --sponsor` which will provide output for each project. If no sponsors are listed, the project will provide a message saying there are no packages looking for sponsors. Finally, an empowering message will be provided regarding the follow-up command to learn more about sponsoring an individual package:
+
+```plaintext
+Project 'Umbraco.Core' has the following package references looking for sponsors
+   [net7.0]: 
+   Top-level Package                                              Requested            Resolved      
+   > Moq                                                          4.18.2               4.18.2
+   > Serilog                                                 	  2.12.0               2.12.0
+   
+The given project `Umbraco.Web.UI` has no packages looking for sponsors given the current sources.
+
+To learn more about sponsoring a package, run `dotnet sponsor package <PACKAGE_NAME>`.
+```
+
+#### dotnet sponsor package <PACKAGE_NAME>
+
+When an individual is feeling generous to sponsor a package, they can run the `dotnet sponsor package <PACKAGE_NAME>` command which will open up the default browser on their machine with the url of the first sponsor defined in the package metadata.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+Details will be provided once the functional proposal is accepted by the .NET OSS community.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+There may be arguments that it isn't the package registry's responsibility to bring awareness to sponsorship needs as it should focus purely as a means to index packages and deliver them reliably to consumers.
+
+It is likely that this functionality will not "move the needle" on OSS sustainability. It is just as likely that there will be negative perception of the excess noise added to individual's development environments where users may go to immediately opt-out of any sponsorship experience. 
+
+On the opposite end, there may be a positive perception of awareness to a largely ignored problem. That individuals can bring more awareness to sponsorship needs and even build tooling on-top of this functionality to empower the .NET OSS community further.
+
+## Rationale and alternatives
```

Comment:
I see two other mechanisms to deliver sponsor metadata from author to consumer:

1. Package owners can set sponsorship information on the package source web UI **at the user or organization level**. You'd set it up once and it would automatically flow to all of your packages that you directly own. We would have a similar "how to display multiple sponsorships" question that your package metadata allows but it comes from the fact that packages can have multiple owners.
2. Package owners can set sponsorship information on the package source web UI **at the package ID** level. There could perhaps be an option to set it on all of the IDs you own, and automatically as new IDs get uploaded for the first time.

These are alternatives to baking the metadata into the .nuspec.

Some questions to help guide the decision:

1. Is sponsorship information immutable per package version? Is it really an immutable concept?
    - I think the answer is most often _no_ because we've faced frustration from package authors about immutable project URL when their DNS lapses or they change their repository location but not the content of the source tree.
1. Do package owners want to advertise sponsorship on some versions but not others?
    - I think probably not. Ideally, it's show on all versions, right? But showing it only on the latest version (or newly published versions) is probably good enough since we know consumers browse latest metadata most often.
1. Do package owners want to have different sponsorship information per package they own? 
    - I don't have an answer for this. The package metadata model allows this, the package ID-level metadata model allows this. The user/org-level metadata model does not allow this.
1. Is there a reasonable way to get server-only metadata on the Installed tab in VS. Said another way, do you want sponsorship information for offline contexts?
    - If we want to show sponsorship on the Installed tab, either we need to fetch and cache it or it need to be inside the package. This begins to feel a lot like the Vulnerability metadata problem where you want to light it up quickly without a million server round trips in restore. In other words, if the sponsor metadata is not in the package (it only lives on the server) we would need to build a high-performance mechanism for `restore` to report all transitive sponsorships.

Taking a server-side metadata option that I suggested may be much cheaper from the implementation perspective. We just need to do the protocol changes to share this new metadata and some NuGet.org UI. But there are drawbacks in the experience as listed above.

Reply:
My research and cross-ecosystem comparisons/talks with other pkg mgmt pms suggests this should all be done inside the confines of a package. Anything registry specific may be challenging to understand and formally document.

That way people can change their sponsors as their project develops. Maybe they always have sponsor data, maybe they meet their goals and remove the metadata in future releases or change it to a cause they would rather promote than themselves. etc. 

Maybe this is a good polling opportunity. Do people want to declare sponsors at pack time or publish time.